app/renderer/reducers/Inspector.js

```diff
@@ -333,6 +336,49 @@ export default function inspector (state = INITIAL_STATE, action) {
         showKeepAlivePrompt: false,
       };
 
+    case SELECT_ACTION_GROUP:
+      return {
+        ...state,
+        selectedActionGroup: action.group
+      };
+
+    case SELECT_SUB_ACTION_GROUP:
+      return {
+        ...state,
+        selectedSubActionGroup: action.group,
+      };
+
+    case SELECT_INTERACTION_MODE:
+      return {
+        ...state,
+        selectedInteractionMode: action.interaction,
+      };
+
+    case ENTERING_ACTION_ARGS:
+      return {
+        ...state,
+        pendingAction: {
+          actionName: action.actionName,
+          action: action.action,
+          args: [],
+        }
+      };
+
+    case SET_ACTION_ARG:
+      return {
+        ...state,
+        pendingAction: {
+          ...state.pendingAction,
+          args: Object.assign([], state.pendingAction.args, {[action.index]: action.value}), // Replace 'value' at 'index'
```

Comment:
What is done here?

Reply:
Makes a copy of `state.pendingAction.args` but replaces the value at `action.index` with `action.value`....

this was the best way I could think of to change a value of an array item immutably.

Reply:
let say we have `const a = [1,2,3,]` then in order to replace the second element with 4 we could `const b = [...a.slice(0, 2), 4, ...s.slice(3)]`